src/scene/materials/standard-node-material.js

```diff
@@ -0,0 +1,100 @@
+import { standard } from '../../graphics/program-lib/programs/standard.js';
+
+import {
+    SHADER_FORWARDHDR, SHADER_PICK
+} from '../constants.js';
+
+import { StandardMaterial } from './standard-material.js';
+
+/**
+ * @private
+ * @class
+ * @name pc.StandardNodeMaterial
+ * @augments pc.StandardMaterial
+ * @classdesc StandardNodeMaterial is sub class of the StandardMaterial, and adds shader graph interop functionality
+ * @param {pc.StandardMaterial} mat - optional material which is cloned
+ * @param {any} chunk - optional shader graph node chunk to be used
+ */
+function StandardNodeMaterial(mat, chunk) {
+    StandardMaterial.call(this);
+
+    if (mat) {
+        StandardMaterial.prototype._cloneInternal.call(mat, this);
+    }
+
+    if (chunk) {
+        this._shaderGraphChunk = chunk;
+    }
+}
+
+StandardNodeMaterial.prototype = Object.create(StandardMaterial.prototype);
+StandardNodeMaterial.prototype.constructor = StandardNodeMaterial;
+
+Object.assign(StandardNodeMaterial.prototype, {
+
+    clone: function () {
```

Comment:
this is public function (overrides Material. ) and should have documentation even if the same, otherwise it won't show up in manual

Reply:
ah - I shall put the JSDoc block back in

Reply:
these blocks are back in - but I had to mark them as `@private` because the class is private